src/sortStudents.ts

```diff
@@ -1,16 +1,63 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const copy = [...students];
+  const callback = (acc: number, current: number): number => {
+    return acc + current;
+  };
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  return copy.sort((a, b) => {
+    switch (sortBy) {
+      case SortType.Name:
+        return order === 'asc'
+          ? a.name.localeCompare(b.name)
+          : b.name.localeCompare(a.name);
+
+      case SortType.Surname:
+        return order === 'asc'
+          ? a.surname.localeCompare(b.surname)
+          : b.surname.localeCompare(a.surname);
+
```

Comment:
you can group some cases

```suggestion
      case SortType.Name:
      case SortType.Surname:
        return order === 'asc'
          ? a[sorrtBy].localeCompare(b[sorrtBy])
          : b[sorrtBy].localeCompare(a[sorrtBy]);

```